### PR TITLE
Add native currency magiclink support

### PR DESCRIPTION
**Pull request by JamesANZ**
Allows anybody to drop the native currency of the network to someone. 

Merge after:
- xdai team is happy to use the contract and payment server (they may make their own changes)
- Payment server is online and ready to interact with the app. This is needed to prevent theft of link data by other nodes on the network

### CODE REVIEW

File: AlphaWallet/Market/MarketQueueHandler.swift [hunk 1]
```diff
@@ -57,10 +57,11 @@ public class MarketQueueHandler {
                 indices: indices.hexa2Bytes.map({ UInt16($0) }),
                 expiry: BigUInt(expiry, radix: 16)!,
                 contractAddress: contractAddress,
-                start: BigUInt(orderObj["start"].string!)!,
-                count: orderObj["count"].intValue,
+                count: BigUInt(orderObj["count"].intValue),
```
[the remark this answers is not in the document]
Comment (James-Sangalli) [reply AlphaWallet/Market/MarketQueueHandler.swift]: Yes, count is the amount of native currency in szabo

File: AlphaWallet/Market/UniversalLinkHandler.swift [hunk 2]
```diff
@@ -216,18 +218,9 @@ public class UniversalLinkHandler {
...
-            messageWithSzabo.append(expiryBytes[i])
-        }
-        for i in 64...83 {
-            messageWithSzabo.append(message[i])
-        }
-        for i in 0..<indices.count {
-            messageWithSzabo.append(indices[i])
-        }
+        messageWithSzabo.append(contentsOf: priceBytes)
+        messageWithSzabo.append(contentsOf: expiryBytes)
```
Comment (hboon) [inline AlphaWallet/Market/UniversalLinkHandler.swift]: Missed out appending message[64...83]

**Comment by James-Sangalli** [comment]
@hboon we could technically merge this and simply put another issue up to implement the payment master, there is no risk to merging as it would not be used in magic links anyway but could be useful for testing in a later build. What do you think?

**Comment by hboon** [comment]
> @hboon we could technically merge this and simply put another issue up to implement the payment master, there is no risk to merging as it would not be used in magic links anyway but could be useful for testing in a later build. What do you think?

@James-Sangalli yes, sounds good. Want to rebase+squash and rebase master first? And then a quick test to see if the original MagicLink and spawnable works unchanged?

**Pull request merged**